### PR TITLE
MAINT: Remove unnecessary checks

### PR DESCRIPTION
**Pull request by larsoner**
1. Restore `has_nibabel` with deprecation so that `mne-bids` users don't have `import mne_bids` fail once we cut a release (https://github.com/mne-tools/mne-bids/pull/1128) -- not worried about `requires_*` that we removed since those are purely testing mechanisms
2. Use `git grep "check_version(['\"].*['\"],.*"` to find occurrences of `check_version(<name>, <ver>)` that are automatically satisfied by our min numpy/scipy/matplotlib.

### CODE REVIEW

File: mne/simulation/tests/test_raw.py [hunk 1]
```diff
@@ -284,10 +284,9 @@ def test_simulate_raw_sphere(raw_data, tmp_path):
     del raw_sim_hann
 
     # check that new Generator objects can be used
```
Comment (drammock) [inline mne/simulation/tests/test_raw.py:286]:
```suggestion
```

File: mne/simulation/tests/test_source.py [hunk 2]
```diff
@@ -237,10 +236,9 @@ def test_simulate_sparse_stc_single_hemi(_get_fwd_labels):
     assert_array_equal(stc_1.rh_vertno, stc_2.rh_vertno)
 
     # smoke test for new API
-    if check_version('numpy', '1.17'):
-        simulate_sparse_stc(fwd['src'], len(labels_single_hemi), times,
-                            labels=labels_single_hemi,
-                            random_state=np.random.default_rng(0))
+    simulate_sparse_stc(fwd['src'], len(labels_single_hemi), times,
+                        labels=labels_single_hemi,
+                        random_state=np.random.default_rng(0))
```
Comment (drammock) [inline mne/simulation/tests/test_source.py:241]: if this was a smoke test for the new numpy rng api, do we even need it at all anymore?

File: mne/stats/tests/test_permutations.py [hunk 3]
```diff
@@ -76,6 +75,5 @@ def test_ci():
                                                   random_state=0),
                     rtol=.1)
     # smoke test for new API
-    if check_version('numpy', '1.17'):
-        random_state = np.random.default_rng(0)
-        bootstrap_confidence_interval(arr, random_state=random_state)
+    random_state = np.random.default_rng(0)
+    bootstrap_confidence_interval(arr, random_state=random_state)
```
Comment (drammock) [inline mne/stats/tests/test_permutations.py:79]: idem

File: mne/tests/test_source_space.py [hunk 4]
```diff
@@ -157,13 +156,12 @@ def test_add_patch_info(monkeypatch):
             add_source_space_distances(src_new, dist_limit=0)
 
     # New SciPy can
```
Comment (drammock) [inline mne/tests/test_source_space.py:158]:
```suggestion
```

File: mne/simulation/tests/test_source.py (hunk 2 above)
Comment (larsoner) [reply mne/simulation/tests/test_source.py:241]: I think it's good to have tests for check_random_state that the new API can be used, I guess we could keep those just in `utils/` though

**Pull request merged**